src/hotspot/share/memory/metaspace/blockTree.cpp

```diff
@@ -36,27 +37,40 @@ namespace metaspace {
 // Needed to prevent linker errors on MacOS and AIX
 const size_t BlockTree::MinWordSize;
 
+#define NODE_FORMAT \
+  "@" PTR_FORMAT \
+  ": canary: " INTPTR_FORMAT \
+  ", parent " PTR_FORMAT \
+  ", left " PTR_FORMAT \
+  ", right " PTR_FORMAT \
+  ", next " PTR_FORMAT \
+  ", size " SIZE_FORMAT
+
+#define NODE_FORMAT_ARGS(n) \
+  p2i(n), \
+  ((n) ? (n)->_canary : 0), \
+  p2i((n) ? (n)->_parent : NULL), \
+  p2i((n) ? (n)->_left : NULL), \
+  p2i((n) ? (n)->_right : NULL), \
+  p2i((n) ? (n)->_next : 0), \
+  ((n) ? (n)->_word_size : 0)
+
 #ifdef ASSERT
 
 // Tree verification
 
-// These asserts prints the tree, then asserts
-#define assrt(cond, format, ...) \
+// This assert prints the tree, then stops (generic message)
+#define assrt0(cond, format, ...) \
```

Comment:
The names get hard to read. I thought it was as typo initially :) Suggestion: `tree_assert`?